src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenOptions.cs

```diff
@@ -23,6 +23,14 @@ public class SwaggerGenOptions
         public List<FilterDescriptor> DocumentFilterDescriptors { get; set; } = new List<FilterDescriptor>();
 
         public List<FilterDescriptor> SchemaFilterDescriptors { get; set; } = new List<FilterDescriptor>();
+
+        public List<FilterDescriptor> ParameterAsyncFilterDescriptors { get; set; } = new List<FilterDescriptor>();
+
+        public List<FilterDescriptor> RequestBodyAsyncFilterDescriptors { get; set; } = new List<FilterDescriptor>();
+
+        public List<FilterDescriptor> OperationAsyncFilterDescriptors { get; set; } = new List<FilterDescriptor>();
+
+        public List<FilterDescriptor> DocumentAsyncFilterDescriptors { get; set; } = new List<FilterDescriptor>();
```

Comment:
Have you investigated whether it's possible to extend the existing functionality to support both types of filters so we don't have to double the surface area?

Reply:
Hi, I pushed a new change that removes those additional lists but that also removes the elegant one-liners with ForEach when creating the filters. I tried pattern matching first, but if a filter implements both sync and async filters, only one will be registered.